plugins/base/frontend/src/main/components/navigationPaneSearch/navigationPaneSearch.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useCallback, useState, useEffect } from 'react';
+import {Select, List } from '@jetbrains/ring-ui';
+import { DokkaFuzzyFilterComponent } from '../search/dokkaFuzzyFilter';
+import { IWindow, Option } from '../search/types';
+import './navigationPaneSearch.scss';
+import ClearIcon from 'react-svg-loader!./clear.svg';
+
+export const NavigationPaneSearch = () => {
+    const defaultWidth = 300
+
+    const [navigationList, setNavigationList] = useState<Option[]>([]);
+    const [selected, onSelected] = useState<Option | null>(null);
+    const [minWidth, setMinWidth] = useState<number>(defaultWidth);
+    const [filterValue, setFilterValue] = useState<string>('')
+    
+    const onChangeSelected = useCallback(
+        (element: Option) => {
+            window.location.replace(`${(window as IWindow).pathToRoot}${element.location}`)
+            onSelected(element);
+        },
+        [selected]
+    );
+
+    const onFilter = (filterValue: string, filteredRecords?: Option[]) => {
+        if(filteredRecords){
+            const requiredWidth = Math.max(...filteredRecords.map(e => e.label.length*9), defaultWidth)
```

Comment:
Could you please extract this magic `9` into constant. It is not obvious why is it here and it can quickly become unmaintainable.